metrics/cloudwatch/cloudwatch.go

```diff
@@ -133,11 +144,41 @@ func (cw *CloudWatch) Send() error {
 		}
 	}
 
-	_, err := cw.svc.PutMetricData(&cloudwatch.PutMetricDataInput{
-		Namespace:  aws.String(cw.namespace),
-		MetricData: datums,
-	})
-	return err
+	var tokens = make(chan struct{}, cw.numConcurrentRequests)
+	var errors = make(chan error)
+	var n int
+
+	for len(datums) > 0 {
+		var batch []*cloudwatch.MetricDatum
+		lim := min(len(datums), cw.numConcurrentRequests)
+		batch, datums = datums[:lim], datums[lim:]
+		n++
+		go func(batch []*cloudwatch.MetricDatum) {
+			tokens <- struct{}{}
+			_, err := cw.svc.PutMetricData(&cloudwatch.PutMetricDataInput{
+				Namespace:  aws.String(cw.namespace),
+				MetricData: batch,
+			})
+			<-tokens
+			errors <- err
```

Comment:
Is there a reason you decrement the semaphore _before_ returning the error along the errors chan? If you didn't need to do that, it would be nicer to express this as

```go
tokens <- struct{}{}
defer func() { <-tokens }()
_, err := cw...
errors <- err
```